src/RomModule.cpp

```diff
@@ -127,9 +127,85 @@ void RomModule::ReadFromRom()
 	addr = PeripheralReadByte(PP_PortB);
 	addr |= (WORD) (PeripheralReadByte(PP_PortC) << 8);
 
-	if (addr & 0x8000)
+	if ((addr & 0x8000) || RomPack == NULL)
 		PeripheralWriteByte(PP_PortA, 0xFF);
 	else
 		PeripheralWriteByte(PP_PortA, *(RomPack + addr));
 }
 //---------------------------------------------------------------------------
+//---------------------------------------------------------------------------
+MegaModule::MegaModule()
+{
+    // debug("MegaModule", "ctor");
+	page = 0;
+	RomPages[0] = new BYTE[ROM_PACK_SIZE];
+	memset(RomPages[0], 0xFF, ROM_PACK_SIZE);
+	RomPack = RomPages[0];
+	for (int i = 1; i < MEGA_MODULE_MAX_PAGES; i++)
+		RomPages[i] = NULL;
+	OnCpuReadA.connect(this, &MegaModule::ReadFromRom);
+}
+//---------------------------------------------------------------------------
+MegaModule::~MegaModule()
+{
+    // debug("MegaModule", "dtor");
+	for (int i = 0; i < MEGA_MODULE_MAX_PAGES; i++) {
+		if (RomPages[i] != NULL) {
+			delete [] RomPages[i];
+			RomPages[i] = NULL;
+		}
+	}
+	RomPack = NULL;
+}
+//---------------------------------------------------------------------------
+void MegaModule::ResetDevice(int ticks)
+{
+    debug("MegaModule", "reset");

```

Comment:
I prefer when debug messages specifically tell even noobs something, if it's possible.